DesktopModules/Vanjaro/UXManager/Extensions/Block/Menu/Controllers/MenuController.cs

```diff
@@ -53,6 +58,29 @@ private static List<StringText> GetPages()
             return Library.Managers.PageManager.GetParentPages(PortalSettings.Current).Where(p => p.TabID != -1).Select(a => new StringText() { Value = a.TabID.ToString(), Text = a.TabName }).ToList();
         }
 
+        private static List<StringText> GetTemplates()
+        {    
+            string TemplatesPath = HttpContext.Current.Server.MapPath("~/Portals/_default/vThemes/" + ThemeManager.CurrentTheme.Name + "/blocks/Menu/Templates");
```

Comment:
Does this need cache? How often is it called? Everytime when page is rendered or only when you're editing a block?